source/cellular_bg770_api.c

```diff
@@ -6475,3 +6500,787 @@ CellularError_t Cellular_SetServiceSelection( CellularHandle_t cellularHandle,
+    {
+        LogError( ( "_isLTEBandMaskNonZero: bad parameter" ) );
+        return false;
+    }
+
+    static const size_t LTE_BAND_MASK_LENGTH = sizeof( pFrequencyBands->asBytes );
+    for( int i = 0; i < LTE_BAND_MASK_LENGTH; i++ )
+    {
+        if( pFrequencyBands->asBytes[ i ] != 0 )
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+CellularError_t Cellular_SetLTEFrequencyBands( CellularHandle_t cellularHandle,
+                                               const CellularLTEBandMask_t *const pFrequencyBands )
+{
+    CellularContext_t * pContext = ( CellularContext_t * ) cellularHandle;
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+    CellularLTEBandMask_t frequencyBandsCopy = { 0 };
+    bool bandsFiltered = false;
+    char frequencyBandsBuffer[ BG770_LTE_BAND_HEX_STRING_MAX_LENGTH + 1 ] = { '\0' };
+    char cmdBuf[ CELLULAR_AT_CMD_MAX_SIZE ] = { '\0' };
+    CellularAtReq_t atReqSetFrequencyBands = { 0 };
+
+    atReqSetFrequencyBands.pAtCmd = cmdBuf;
+    atReqSetFrequencyBands.atCmdType = CELLULAR_AT_NO_RESULT;
+    atReqSetFrequencyBands.pAtRspPrefix = NULL;
+    atReqSetFrequencyBands.respCallback = NULL;
+    atReqSetFrequencyBands.pData = NULL;
+    atReqSetFrequencyBands.dataLen = 0;
+
+    cellularStatus = _Cellular_CheckLibraryStatus( pContext );
+
+    if( cellularStatus != CELLULAR_SUCCESS )
+    {
+        LogError( ( "_Cellular_CheckLibraryStatus failed" ) );
+    }
+    else if( ( pFrequencyBands == NULL ) || ( !_isLTEBandMaskNonZero( pFrequencyBands ) ) )
+    {
+        LogError( ( "Cellular_SetLTEFrequencyBands : Bad parameter" ) );
+        cellularStatus = CELLULAR_BAD_PARAMETER;
+    }
+    else
+    {
+        static const CellularLTEBandMask_t SUPPORTED_BG770_LTE_BAND_MASK = {
+                // 0x2000000000f0e189f
+                .asBytes = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00, 0x0f, 0x0e, 0x18, 0x9f }
+            };
+
+        memcpy(&frequencyBandsCopy, pFrequencyBands, sizeof( frequencyBandsCopy ) );
+
+        /* Limit to frequency bands supported by Quectel. */
+
+        if( !_filterLTEBandMask( &frequencyBandsCopy, &SUPPORTED_BG770_LTE_BAND_MASK, &bandsFiltered ) )
+        {
+            LogError( ( "Cellular_SetLTEFrequencyBands : Failed to filter frequency bands to BG770 supported" ) );
+            cellularStatus = CELLULAR_INTERNAL_FAILURE;
+        }
+
+        if( cellularStatus == CELLULAR_SUCCESS )
+        {
+            if( !_isLTEBandMaskNonZero( pFrequencyBands ) )
+            {
+                LogError( ( "Cellular_SetLTEFrequencyBands : Specified frequency bands not supported by BG770" ) );
+            }
+            else
+            {
+                if( bandsFiltered )
+                {
+                    LogWarn( ( "Cellular_SetLTEFrequencyBands : Unsupported LTE frequency bands removed" ) );
+                }
+            }

```

Comment:
`else if` or is this a MISRA thing?

Reply:
Not a MISRA thing, just changed structure and forgot to consolidate